tools/state-viewer/src/commands.rs

```diff
@@ -1057,6 +1062,148 @@ pub(crate) fn clear_cache(store: Store) {
     store_update.commit().unwrap();
 }
 
+#[derive(PartialEq, Eq)]
+pub struct StateStatsStateRecord {
+    key: Vec<u8>,
+    value: Vec<u8>,
+}
+
+impl StateStatsStateRecord {
+    pub fn size(&self) -> ByteSize {
+        ByteSize::b(self.key.len() as u64 + self.value.len() as u64)
+    }
+}
+
+impl Ord for StateStatsStateRecord {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.size().cmp(&other.size()).reverse()
+    }
+}
+
+impl PartialOrd for StateStatsStateRecord {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl std::fmt::Debug for StateStatsStateRecord {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let state_record = StateRecord::from_raw_key_value(self.key.clone(), self.value.clone());
+
+        let Some(state_record) = state_record else { return None::<StateRecord>.fmt(f) };
+
+        f.debug_struct("StateStatsStateRecord")
+            .field("account_id", &state_record_to_account_id(&state_record).as_str())
+            .field("type", &state_record.get_type_string())
+            .field("size", &self.size())
+            .finish()
+    }
+}
+
+#[derive(Default)]
+pub struct StateStats {
+    pub total_size: ByteSize,
+    pub total_count: usize,
+
+    pub top_state_records: BinaryHeap<StateStatsStateRecord>,
+
+    pub middle_state_record: Option<StateStatsStateRecord>,
+}
+
+impl core::fmt::Debug for StateStats {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let average_size = self
+            .total_size
+            .as_u64()
+            .checked_div(self.total_count as u64)
+            .map(ByteSize::b)
+            .unwrap_or_default();
+        f.debug_struct("StateStats")
+            .field("total_size", &self.total_size)
+            .field("total_count", &self.total_count)
+            .field("average_size", &average_size)
+            .field("middle_state_record", &self.middle_state_record.as_ref().unwrap())
+            .field("top_state_records", &self.top_state_records)
+            .finish()
+    }
+}
+
+impl StateStats {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn push(&mut self, key: Vec<u8>, value: Vec<u8>) -> () {
+        self.total_size += key.len() as u64 + value.len() as u64;
+        self.total_count += 1;
+
+        self.top_state_records.push(StateStatsStateRecord { key, value });
+        if self.top_state_records.len() > 5 {
+            self.top_state_records.pop();
+        }
+    }
+}
+
+fn read_flat_state_value(
```

Comment:
I don't think the value itself is used anywhere... Quick optimization, we can perhaps just work with the size of the value instead of the vector value itself?